### PR TITLE
feat: lowest-direct resolution mode

### PR DESCRIPTION
**Pull request by zkochan**
This is going to be the default resolution mode in v8.

### CODE REVIEW

**Review by BlackHole1 — APPROVED** [review]
LGTM

File: .changeset/pretty-adults-breathe.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@pnpm/resolve-dependencies": minor
+"@pnpm/core": minor
+"@pnpm/config": minor
+"pnpm": minor
+---
+
+A new `resolution-mode` added: `lowest-direct`. With this resolution mode direct dependencies will be resolved to their lowest versions. So if there is `foo@^1.1.0` in the dependencies, then `1.1.0` will be installed, even if the latest version of `foo` is `1.2.0`.
```
Comment (bvanjoi) [inline .changeset/pretty-adults-breathe.md:8]: It means the subdependencies still use the target same as "highest"?

Comment (zkochan) [reply .changeset/pretty-adults-breathe.md:8]: yes

**Comment by zkochan** [comment]
Package managers do some deduplication during installation. So they will not always install the latest version that matches the range. Sometimes they will install the lowest version. It is better to test your changes with the lowest possible version.

**Pull request merged**